src/hooks/useTransactionNonces.ts

```diff
@@ -51,12 +51,9 @@ const useTransactionNonces = ({ transaction }: Props) => {
   const { latestReceivedNonceMethodName } = getSubstrateDynamicNames(sourceChain);
   const { api: targetApi } = getValuesByChain(targetChain);
 
-  useEffect(() => {
-    if (!areApiLoading || !transaction || !transaction || isTransactionCompleted(transaction)) {
-      return;
-    }
-    const getNonceByHash = async () => {
-      const blockHash = await targetApi.rpc.chain.getBlockHash(bestBlockFinalizedOnTarget);
+  const getNonceByHash = useCallback(
```

Comment:
3) In the past i was calling this function every time `bestBlockFinalizedOnTarget` got updated. As now i need to call this function specifically  i obtain the block of the delivered message event, i had to expose it to the parent hook so this process can be run at that time.

Reply:
I'm a bit confused, it's still being called every time `bestBlockFinalizedOnTarget` changes, cause you are invoking that function in `useEffect`. Am I missing something?

Reply:
Also the fucntion should be rather be called `getLatestReceivedNonceByHash`, cause now one might wonder what kind of nonce is this.

Reply:
I'm calling that function twice :

1) one for getting the message nonce related to the specific block i needed 
2) Then i'm calling it every time a `bestBlockFinalizedOnTarget` arrives  so i can match it with the step 1) and infer that step is completed.

Anyway, now i realize that i might avoid the step 1 by improving the logic.
Also mentioning  this in the issue #159